### PR TITLE
Sink: GCP Pub/Sub

### PR DESCRIPTION
**Pull request by ehazlett**
This adds a new sink for sending to the GCP Pub/Sub service.

### CODE REVIEW

File: pkg/sinks/gcppubsub/gcppubsub.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	return nil
+}
+
+func (s *GCPPubSub) Send(ctx context.Context, payload *kt.Output) {
+	go s.send(ctx, payload.Body)
+}
+
+func (s *GCPPubSub) send(ctx context.Context, payload []byte) {
+	res := s.topic.Publish(ctx, &pubsub.Message{Data: payload})
```
Comment (i3149) [inline pkg/sinks/gcppubsub/gcppubsub.go:72]: I thought that gsub has keys in addition to data but I guess not.

File: pkg/sinks/gcppubsub/gcppubsub.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	s.client = client
+	s.topic = s.client.Topic(s.topicID)
+
+	s.Infof("connected to GCP PubSub for project %s on topic %s", s.projectID, s.topicID)
+
+	return nil
+}
+
+func (s *GCPPubSub) Send(ctx context.Context, payload *kt.Output) {
+	go s.send(ctx, payload.Body)
```
Comment (i3149) [inline pkg/sinks/gcppubsub/gcppubsub.go:68]: I use this pattern everywhere but I worry that in an extreme case there will be too many goroutines running. I don't think worth worrying about it now.

File: pkg/sinks/gcppubsub/gcppubsub.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func (s *GCPPubSub) send(ctx context.Context, payload []byte) {
+	res := s.topic.Publish(ctx, &pubsub.Message{Data: payload})
+	if _, err := res.Get(ctx); err != nil {
```
Comment (i3149) [inline pkg/sinks/gcppubsub/gcppubsub.go:73]: Could pass res to another channel and process one at a time but probably not worth it.

**Review by i3149 — APPROVED** [review]
LGMT. There's also a gcloud sink. Do you think its worth keeping this? No one uses it as far as I know.

File: pkg/sinks/gcppubsub/gcppubsub.go (hunk 2 above)
Comment (ehazlett) [reply pkg/sinks/gcppubsub/gcppubsub.go:68]: I thought about this too but I think we can revisit later.  Maybe we can add some profiling as well to get some more data.

File: pkg/sinks/gcppubsub/gcppubsub.go (hunk 3 above)
Comment (ehazlett) [reply pkg/sinks/gcppubsub/gcppubsub.go:73]: I think in this case it's just to confirm it succeeded.  If we wanted to process perhaps we could add a separate channel.

Comment (i3149) [reply pkg/sinks/gcppubsub/gcppubsub.go:73]: Makes sense. I think Publish is non blocking but Get will block until the request works/fails. Fine as is.

**Pull request merged**